processing/src/main/java/io/druid/segment/IndexIO.java

```diff
@@ -748,7 +808,7 @@ public int size()
             channel.write(ByteBuffer.wrap(specBytes));
             serdeficator.write(channel);
             channel.close();
-          } else if (filename.startsWith("met_")) {
+          } else if (filename.startsWith("met_") || filename.startsWith("numeric_dim_")) {
```

Comment:
My thinking in this area was that the V8 segment format/legacy IndexMerger would be deprecated by the time we really have dimension types beyond long/float, i.e., this code would no longer be relevant when it's time to extend it further. (could take the stance that users who want extra dimension types must use buildV9directly)

So this was a quick way for IndexIO's convertV8toV9() to be able to distinguish between dimensions that use a dictionary encoded String style column, or ones that use the metric-style.